### PR TITLE
Task2

### PR DESCRIPTION
**Pull request by KaroukinStanislau**

### CODE REVIEW

File: task2/index.html [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      <figcaption>LADA</figcaption>
+    </figure>
+  </aside>
+
+  <footer>
+    Footer Contact information: <a href="mailto:someone@example.com">
+      someone@example.com</a>.
+  </footer>
+
+  <script type="text/javascript" src="script.js"></script>
```
Comment (NikitaSedyx) [inline task2/index.html]: `./`

File: task2/script.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+function isArrayTwo(arr) {
+  return arr instanceof Array;
+}
+
+function isArrayThree(arr) {
+  return {}.toString.call(arr) === '[object Array]'
+}
+
+function range() {
```
Comment (NikitaSedyx) [inline task2/script.js]: where are arguments?

File: task2/script.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      {
+        for (var i = arguments[0]; i < arguments[1]; i++) {
+          arr.push(i);
+        }
+      }
+      break;
+    case 3:
+      {
+        for (var i = arguments[0]; i < arguments[1]; i++) {
+          if (i % (arguments[2]) !== 0) {
```
Comment (NikitaSedyx) [inline task2/script.js]: what does it do?
the third argument is step

File: task2/script.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  return arr instanceof Array;
+}
+
+function isArrayThree(arr) {
+  return {}.toString.call(arr) === '[object Array]'
+}
+
+function range() {
+  var arr = [];
+  switch (arguments.length) {
```
Comment (NikitaSedyx) [inline task2/script.js]: not cool switch usage, if is better here
you have mostly the same loop, why do you copy past it?

File: task2/script.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        }
+      }
+      break;
+  }
+  return arr;
+}
+
+function compact(arr) {
+  return arr.filter(function(arr) {
+    return Boolean(arr).valueOf() === true;
```
Comment (NikitaSedyx) [inline task2/script.js]: looks like I see the same second time
please read about filter, falsy values, type converting

File: task2/script.js [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    return array.indexOf(element) === index;
+  });
+}
+
+function last(arr) {
+  return arr[(arr.length - 1)];
+}
+
+function excludeLast(arr, count = 1) {
+  arr.length = arr.length - count;
```
Comment (NikitaSedyx) [inline task2/script.js]: it mutates array, -1

**Review by NikitaSedyx — CHANGES_REQUESTED** [review]
js is mostly the same with @RamanKreiza
do you work together?

File: task2/script.js [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+function isArrayOne(arr) {
+  return arr.constructor === Array;
+}
+
+function isArrayTwo(arr) {
+  return arr instanceof Array;
+}
+
+function isArrayThree(arr) {
+  return {}.toString.call(arr) === '[object Array]'
```
Comment (NikitaSedyx) [inline task2/script.js]: are you sure it works?
`SyntaxError: expected expression, got '.'` is in my console

File: task2/script.js [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+function range(a, b, step = 1) {
+  var arr = [];
+  // condition for one argument
+  if (typeof b == 'undefined') {
```
Comment (NikitaSedyx) [inline task2/script.js]: `b === undefined`

File: task2/script.js [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+    b = a;
+    a = 0;
+  }
+  for (var i = a; i < b; i += step) {
```
Comment (NikitaSedyx) [inline task2/script.js]: could you run `range(10, 20, -5)`?

File: task2/script.js [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+    arr.push(i);
+  }
+  return arr;
+}
+
+function compact(arr) {
+  return arr.filter(function(element) {
+    return element ? true : false;
```
Comment (NikitaSedyx) [inline task2/script.js]: why do you need it? please read about converting in conditions

File: task2/script.js [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+  });
+}
+
+function compactCycle(arr) {
+  var res = [];
+  for (var i = 0; i < arr.length; i++) {
+    if (arr[i]) {
```
Comment (NikitaSedyx) [inline task2/script.js:46]: please look above and understand

File: task2/script.js [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+function unique(arr) {
+  return arr.filter(function(element, index, array) {
+    return array.indexOf(element) === index;
+  });
+}
+
+function last(arr) {
+  return arr[(arr.length - 1)];
```
Comment (NikitaSedyx) [inline task2/script.js]: why do you need `()`?

File: task2/script.js [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+function excludeLast(arr, count = 1) {
```
Comment (NikitaSedyx) [inline task2/script.js]: es6 feature

**Review by NikitaSedyx — CHANGES_REQUESTED** [review]

File: task2/script.js (hunk 7 above)
Comment (KaroukinStanislau) [reply task2/script.js]: it works for me in chrome 56
https://learn.javascript.ru/class-instanceof
don't work in console. try "( {}.toString.call([]) )"

Comment (NikitaSedyx) [reply task2/script.js]: I'd suggest you to use Object.prototype without any hacks
or wrap your expression
my chrome is 53, you should support not only 1 version of browsers

File: task2/script.js (hunk 10 above)
Comment (KaroukinStanislau) [reply task2/script.js]: finally i got it :)

File: task2/script.js [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+
+function isArrayTwo(arr) {
+  return arr instanceof Array;
+}
+
+function isArrayThree(arr) {
+  return Object.prototype.toString.call(arr) === '[object Array]'
+}
+
+function range(a, b, step) {
```
Comment (NikitaSedyx) [inline task2/script.js:15]: _.range(10, 2, -5) -> [10, 5]

File: task2/script.js [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+  return arr.filter(function(element, index, array) {
+    return array.indexOf(element) === index;
+  });
+}
+
+function last(arr) {
+  return arr[arr.length - 1];
+}
+
+function excludeLast(arr, count) {
```
Comment (NikitaSedyx) [inline task2/script.js:77]: mostly is works

**Review by NikitaSedyx — CHANGES_REQUESTED** [review]

**Review by NikitaSedyx — APPROVED** [review]

**Pull request merged**